### PR TITLE
Upload snapshot of tests in 2 steps

### PR DESCRIPTION
**Pull request by nulls**
Now we upload tests using `commitId` and after it upload it them again using `version` provided by user.
Current implementation copied snapshot to new version (we duplicate content)

### What's done:
* added `TestsSourceSnapshotInfo`
* refactored upload tests

It's part of #1717

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt [hunk 1]
```diff
@@ -24,41 +27,62 @@ import reactor.kotlin.core.publisher.toMono
...
+        val key = TestSuitesSourceSnapshotKey(
+            organizationName = snapshotInfo.organizationName,
+            testSuitesSourceName = snapshotInfo.sourceName,
+            version = snapshotInfo.commitId,
+            creationTime = snapshotInfo.commitTime,
+        )
         val content = part.content().map { it.asByteBuffer() }
-        testSuitesSourceVersionService.upload(versionInfo, content).map { writtenBytes ->
-            log.info { "Saved ($writtenBytes bytes) snapshot of ${versionInfo.sourceName} in ${versionInfo.organizationName} with version ${versionInfo.version}" }
+        snapshotStorage.upload(key, content).map { writtenBytes ->
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt:53]: Could we move such logging into the storage class directly? I think we will need it for any type of `upload/copy/etc`

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt [hunk 2]
```diff
@@ -6,19 +6,13 @@ import kotlinx.serialization.Serializable
 typealias TestsSourceVersionInfoList = List<TestsSourceVersionInfo>
 
 /**
- * @property organizationName
- * @property sourceName
+ * @property snapshotInfo
  * @property version
```
Comment (kgevorkyan) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: Could you please document it

We have `commitId` which is actually `commitHash`, right? And what about `version` from now? What's the meaning of it

Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: `commitId` is `commitHash` -- an unique id of snapshot (tests)
`version` is human readable format provided by user: `tag`, `branch` or `short commit id`. 
We will support fetching by human readable format and creating a snapshot with `commitId` for internal usage and `version` to show for user to select for running a test

Will extend KDoc

Comment (kgevorkyan) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: But, `version` from `TestSuitesSourceSnapshotKey` is still commit id, right?

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt (hunk 2 above)
Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: > But, `version` from `TestSuitesSourceSnapshotKey` is still commit id, right?

Not **still**: previously it was a version provided by user.
Now we will have two `TestSuitesSourceSnapshotKey`: one detected by preprocessor with `long commit id`, second -- with version provided by user

Comment (kgevorkyan) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: Oh, I see, ok

Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: a note: it's a temporary till migration is finished.
We will remove `TestSuitesSourceSnapshotKey` and will use two entities:
- `TestsSourceSnapshot`: as a key to store in storage
- `TestsSourceVersion`: as a key\info for user. Will be stored only in database

Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/test/TestsSourceVersionInfo.kt:10]: > Now we will have two `TestSuitesSourceSnapshotKey`

forgot to copy existed `TestSuites` -- added

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt:53]: moved

**Review by kgevorkyan — APPROVED** [review]

**Pull request merged**